python/lsst/meas/deblender/sourceDeblendTask.py

```diff
@@ -307,15 +307,17 @@ def deblend(self, exposure, srcs, psf):
             minChildren, maxChildren = self.config.ciDeblendChildRange
             nPeaks = np.array([len(src.getFootprint().peaks) for src in srcs])
             childrenInRange = np.where((nPeaks >= minChildren) & (nPeaks <= maxChildren))[0]
-            if len(childrenInRange) < self.config.ciNumParentsToDeblend:
-                raise ValueError("Fewer than ciNumParentsToDeblend children were contained in the range "
-                                 "indicated by ciDeblendChildRange. Adjust this range to include more "
-                                 "parents.")
+            numParentsToDeblend = self.config.ciNumParentsToDeblend
+            if len(childrenInRange) < numParentsToDeblend:
+                # Only use the top N instead of the full requested list.
+                if len(childrenInRange) == 0:
+                    raise ValueError("No children to deblend; cannot continue with CI testing.")
```

Comment:
This is fine but I'm also fine with taking Dan's full suggestion that this isn't the place to check for whether or not there are any blends. So you could just warn here and proceed with no sources to deblend, since there could be other visits (in other bands) that are deblended. Nothing says that there have to be any blends in a visit.

Reply:
I have no idea how to do that, and really need to get this out the door.  I _think_ this is better than what was there before.

Reply:
Sorry if I wasn't clear, I meant instead of raising here I would just issue a warning.

Reply:
Even that has some challenges because I'm not sure how to fix downstream in this task if there really were zero.  I think this would be impossible anyway (call the deblender with no sources or something?)

Reply:
Ok. Like I said, this is fine and if you want to push forward with it you can. But if we come up against instances where even this constraint is too stringent then we can make it a warning later.